src/Backend/OnboardingInsuranceAPI/Areas/Insurance/GetContribution.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.Enums;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class GetContribution : IHandler
+{
+    private readonly DataContext _context;
+
+    public GetContribution(DataContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<ContributionData> Handle(ContributionData requestedData, string pid)
+    {
+        if(string.IsNullOrEmpty(pid))
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+
+        var birthDate = _context.Users.FirstOrDefault(u => u.Pid == pid).Birthdate;
+        if(string.IsNullOrEmpty(birthDate))
+            throw new ApiException(ErrorCode.NotFound);
+
+        var age = DateTime.Now.Year - DateTime.Parse(birthDate).Year;
```

Comment:
Make sure you handled Time Zones well here. Also why not save birthdate as a DateTime or OnlyDate data type straight into database? This parse can possibly throw an exception .

Reply:
Ok, I am planned to change in the next branch datatype from string to DateTime or OnlyDate and I will.